pkg/cmd/pulumi/deployment.go

```diff
@@ -66,6 +67,81 @@ func newDeploymentCmd() *cobra.Command {
 		"Override the file name where the deployment settings are specified. Default is Pulumi.[stack].deploy.yaml")
 
 	cmd.AddCommand(newDeploymentSettingsCmd())
+	cmd.AddCommand(newDeploymentRunCmd())
+
+	return cmd
+}
+
+func newDeploymentRunCmd() *cobra.Command {
+	// Flags for remote operations.
+	remoteArgs := RemoteArgs{}
+
+	var stack string
+	var suppressPermalink string
+
+	cmd := &cobra.Command{
+		Use:   "run <operation> [url]",
+		Short: "Launch a deployment job on Pulumi Cloud",
+		Long:  "",
+		Args:  cmdutil.RangeArgs(1, 2),
+		Run: cmdutil.RunResultFunc(func(cmd *cobra.Command, args []string) result.Result {
+			ctx := cmd.Context()
+
+			operation, err := apitype.ParsePulumiOperation(args[0])
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			var url string
+			if len(args) > 1 {
+				url = args[1]
+			}
+
+			display := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+				// we only suppress permalinks if the user passes true. the default is an empty string
+				// which we pass as 'false'
+				SuppressPermalink: suppressPermalink == "true",
+			}
+
+			project, _, err := readProject()
+			if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
+				return result.FromError(err)
+			}
+
+			currentBe, err := currentBackend(ctx, project, display)
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			if !currentBe.SupportsDeployments() {
+				return result.FromError(fmt.Errorf("backends of this type %q do not support deployments",
+					currentBe.Name()))
+			}
+
+			s, err := requireStack(ctx, stack, stackOfferNew|stackSetCurrent, display)
+			if err != nil {
+				return result.FromError(err)
+			}
+
+			if errResult := validateDeploymentFlags(url, remoteArgs); errResult != nil {
+				return errResult
+			}
+
+			return runDeployment(ctx, display, operation, s.Ref().FullyQualifiedName().String(), url, remoteArgs)
+		}),
+	}
+
+	// Remote flags
+	remoteArgs.applyFlagsForDeploymentCommand(cmd)
+
+	cmd.PersistentFlags().StringVar(
```

Comment:
yes, but it is already this way in `pulumi up` and I left as a string bar for consistency (if people is already used to use it that way it would be weird). Happy to change it to BoolVar tho
 https://github.com/pulumi/pulumi/blob/f69f1d2ee3cdc98c56aaad0094c5b474a1a062df/pkg/cmd/pulumi/up.go#L627-L629

Reply:
ahh I missed the usage of NoOptDefVal, yes I will change to BoolVar then!